frodo.js

```diff
@@ -1,110 +1,103 @@
-const lastElement = array[array.length - 1];
```

Comment:
SCRIPT-You did a great job handling all the possible error states in the UI. It enhances the user experience significantly.